src/main/java/org/jitsi/videobridge/cc/vp8/VP8AdaptiveTrackProjectionContext.java

```diff
@@ -192,23 +200,25 @@ VP8FrameProjection createVP8FrameProjection(
         int payloadOff = rtpPacket.getPayloadOffset();
         if (!DePacketizer.VP8PayloadDescriptor.isStartOfFrame(buf, payloadOff))
         {
+            storeMaxSequenceNumberOfFrame(rtpPacket);
             return null;
         }
 
+        long nowMs = System.currentTimeMillis();
+
         // Lastly, check whether the quality of the frame is something that we
         // want to forward. We don't want to be allocating new objects unless
         // we're interested in the quality of this frame.
-        if (!vp8QualityFilter.acceptFrame(rtpPacket, targetIndex))
+        if (!vp8QualityFilter.acceptFrame(rtpPacket, targetIndex, nowMs))
         {
             return null;
         }
 
-        long nowMs = System.currentTimeMillis();
-
         // We know we want to forward this frame, but we need to make sure it's
```

Comment:
does `this frame` here refer to lastVP8FrameProject? or nextVP8FrameProject?

Reply:
This frame can only refer to nextVP8FrameProject because the lastVP8FrameProjection is already (in the process of being) forwarded.

Reply:
ok, so this is saying "since we are in `createVP8FrameProjection`, it's a given we'll want to forward the frame (projection) we're creating here, but we need to make sure it will produce a decodable stream (which is basically saying, we need this frame to be 'complete', I guess?)  so we'll need to check that creating a projection from the last frame returns a valid result" ?

Reply:
> it's a given we'll want to forward the frame (projection) we're creating here
we're at least interested to forward it, it has passed for example the quality test.

As you say it is possible that the `createVP8FrameProjection` returns null, in which case we don't forward the frame. A reason for the createVP8FrameProjection to return null is that it believes the new frame won't be decodable at the receiver.